clients/python/rostam/kv.py

```
@@ -352,6 +352,130 @@ def exists(self, collection: str, id: int) -> bool:
+        flags = (0x01 if with_vector else 0) | (0x02 if with_payload else 0)
+        payload = self._c._call("vector_get_batch",
+                                _vecwire.encode_vector_get_batch_args(collection, list(ids), flags))
+        rows = _vecwire.decode_get_batch_result(payload or b"\x00\x00\x00\x00")
+        # Lift stored content out of the reserved $content key, mirroring get()'s
+        # Point shape (a not-found row's metadata is already {}, so pop is a no-op).
+        for row in rows:
+            meta = row.get("metadata") or {}
+            row["content"] = meta.pop("$content", "")
+        return rows
+
+    def scroll(self, collection: str, *, filter: Optional[Dict[str, Any]] = None,
```

Ruff (0.16.1)

[error] Function argument `filter` is shadowing a Python builtin

(A002)